### ISSUE TITLE
<dialog-outlet> seems to break the ability to properly go "back".

### ISSUE THREAD

**Issue by mgiuliani** [mseemann/angular2-mdl#147]
Hey there - updated to 2.0.0 yesterday for the dialog provider support (Awesome!).  However, we quickly noticed some issues with the back button after adding the <dialog-outlet> component to our page.  Removing the component restored functionality.

After some digging in our app, we also discovered that this same problem exists in the sample application here.  If you navigate around the demo site without hitting the "dialog" pages, then the browser's back functionality works just fine.  However, once you hit a page with a dialog, the URL no longer updates correctly, and it takes multiple "back" actions to update the view.
1. Open the demo application
2. Navigate to:  Themes, Buttons, Cards.  Go back, back, back.  [OK].
3. Refresh
4. Navigate to: Themes, Buttons, Cards, Dialog (Declarative). Go back.
   - Notice the view changed to Cards, but the URL did not update.
   - Go back again.  The view and URL are now in sync.
   - Go back again.  View changed to buttons, but the URL did not update.
   - Etc.

**Comment by mseemann** [comment]
thx, I'll check this...

**Comment by mseemann** [comment]
right! There is a problem with the Router. If you enable tracing: `RouterModule.forRoot(appRoutes, {enableTracing: true}),` you'll see that the navigation events are doubled. if you go back from the dialog a NavigationCancel Event fires with an error like: `Navigation ID 6 is not equal to the current navigation id 7`. I need to investigate this further...


Workaround:
remove all dialog-outlet elements and do the following in your Root-App-Component (the one that is bootstraped):

```
 constructor(
    private dilalogOuletService: MdlDialogOutletService, 
    private viewConatinerRef: ViewContainerRef) {
    this.dilalogOuletService.setDefaultViewContainerRef(this.viewConatinerRef);
  }
```

**Comment by mgiuliani** [comment]
Aha, nice find.  I was trying to figure out how to enable debug output for the router, so thanks for the tip there :)

I had tried to use the dialogOutletService.setDefaultViewContainerRef mechanism as an alternative, but it doesn't seem that the MdlDialogOutletService is exported anywhere to be pulled in.  I'll dig a bit more -  since it is working on your end, it is likely just an miss on my part..

**Comment by mseemann** [comment]
I have updated the master and demo with this workaround - just in case you want to double check something.

**Comment by mseemann** [comment]
https://github.com/angular/angular/issues/12282

**Comment by mseemann** [comment]
pull request (https://github.com/angular/angular/pull/12330) created that fix this issue in the angular router

**Comment by mseemann** [comment]
the fix (angular/angular#12338) has landed in angular router 3.2.0 - the workaround is no longer needed.
